src/sidebar/components/test/login-control-test.js

```diff
@@ -1,115 +1,284 @@
 'use strict';
 
 var angular = require('angular');
+var proxyquire = require('proxyquire');
 
+var bridgeEvents = require('../../../shared/bridge-events');
 var util = require('../../directive/test/util');
 
-function PageObject(element) {
-  this.unknownLoginText = function () {
-    return element[0].querySelector('.login-text').textContent;
-  };
-  this.loginText = function () {
-    return Array.from(element[0].querySelectorAll('.login-text a'))
-      .map(function (el) { return el.textContent; });
-  };
-  this.menuLinks = function () {
-    return Array.from(element[0].querySelectorAll('.login-control-menu .dropdown-menu a'))
-      .map(function (el) { return el.textContent; });
-  };
-  this.disabledMenuItems = function () {
-    return Array.from(element[0].querySelectorAll('.login-control-menu .dropdown-menu__link--disabled'))
-      .map(function (el) { return el.textContent.trim(); });
-  };
-  this.menuText = function () {
-    return element[0].querySelector('span').textContent;
+function pageObject(element) {
+  return {
+    menuLinks: function () {
+      return Array.from(element[0].querySelectorAll('.login-control-menu .dropdown-menu a'))
+                  .map(function (el) { return el.textContent; });
+    },
+    menuText: function () {
+      return element[0].querySelector('span').textContent;
+    },
+    userProfileButton: element[0].querySelector('.dropdown-menu__user-profile-btn'),
```

Comment:
I think `js-*` classes are better than tests finding elements based on things like text content. It makes the tests easier to write correctly, less likely to contain mistakes (e.g. that cause a test to still pass when the code is wrong), less brittle (less likely to be broken by code changes that don't really break the thing they're intended to test) and easier to read.

In this case there were no existing styled classes that are unique to these elements, so I'll change the newly added ones to `js-*` classes